server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -0,0 +1,394 @@
+    public static final String INDICES_EXTENSION_NAME_ACTION_NAME = "indices:internal/name";
+    public static final String REQUEST_EXTENSION_CLUSTER_STATE = "internal:discovery/clusterstate";
+    public static final String REQUEST_EXTENSION_LOCAL_NODE = "internal:discovery/localnode";
+    public static final String REQUEST_EXTENSION_CLUSTER_SETTINGS = "internal:discovery/clustersettings";
+
+    private static final Logger logger = LogManager.getLogger(ExtensionsOrchestrator.class);
+
+    /**
+     * Enum for Extension Requests
+     *
+     * @opensearch.internal
+     */
+    public static enum RequestType {
+        REQUEST_EXTENSION_CLUSTER_STATE,
+        REQUEST_EXTENSION_LOCAL_NODE,
+        REQUEST_EXTENSION_CLUSTER_SETTINGS,
+        CREATE_COMPONENT,
+        ON_INDEX_MODULE,
+        GET_SETTINGS
+    };
+
+    private final Path extensionsPath;
+    final List<DiscoveryExtensionNode> uninitializedExtensions;
+    List<DiscoveryExtensionNode> extensions;
+    TransportService transportService;
+    ClusterService clusterService;
+
+    public ExtensionsOrchestrator(Settings settings, Path extensionsPath) throws IOException {
+        logger.info("ExtensionsOrchestrator initialized");
+        this.extensionsPath = extensionsPath;
+        this.transportService = null;
+        this.uninitializedExtensions = new ArrayList<DiscoveryExtensionNode>();
+        this.extensions = new ArrayList<DiscoveryExtensionNode>();
+        this.clusterService = null;
+
+        /*
+         * Now Discover extensions
+         */
+        discover();
+
+    }
+
+    public void setTransportService(TransportService transportService) {
+        this.transportService = transportService;
+        registerRequestHandler();
+    }
+
+    public void setClusterService(ClusterService clusterService) {
+        this.clusterService = clusterService;
+    }
+
+    private void registerRequestHandler() {
+        transportService.registerRequestHandler(
+            REQUEST_EXTENSION_CLUSTER_STATE,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            ((request, channel, task) -> channel.sendResponse(handleExtensionRequest(request)))
+        );
+        transportService.registerRequestHandler(
+            REQUEST_EXTENSION_LOCAL_NODE,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            ((request, channel, task) -> channel.sendResponse(handleExtensionRequest(request)))
+        );
+        transportService.registerRequestHandler(
+            REQUEST_EXTENSION_CLUSTER_SETTINGS,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            ((request, channel, task) -> channel.sendResponse(handleExtensionRequest(request)))
+        );
+    }
+
+    @Override
+    public PluginsAndModules info() {
+        return null;
```

Comment:
This should not return `null` I suppose?

Reply:
Its mostly how reporting is done. In early days of developing extensions we inherited this from PluginsService.java. 

@ryanbogan can you remove `implements ReportingService<PluginsAndModules>` from `ExtensionsManager` class and remove this `@Override`? 
Also open an issue to track reporting of extensions info and we could make it a simple enhancement.